src/IP.php

```diff
@@ -283,4 +419,16 @@ public function isPublic()
     {
         return !$this->isPrivate();
     }
+
+    /**
+     * Ensures that a given $ip is within the range of a valid IPvX address.
+     *
+     * @param \GMP|resource $ip a GMP object or resource
+     *
+     * @return bool
+     */
+    private static function isInRange($ip)
```

Comment:
I find "isInRange" to be a confusing name, especially with your other modification about private ranges (at first glance I thought they are related because of the name, but they are not). What about something like "isValid" or "isInValidRange" for example?

Also here again, what is the reason for making this private?

Reply:
I guess this should be at least protected.